### ISSUE TITLE
[Feature Request]: ui5-input - Support for display value

### ISSUE THREAD

**Issue by vasicvuk** [SAP/ui5-webcomponents#7964]
### Feature Request Description

I wanted to implement a decimal format for the Input component, I managed to implement behavior similar to:

https://docs.telerik.com/blazor-ui/components/numerictextbox/overview

The idea is to have thousand separator displayed on blur and support multiple locales. And I managed to do so, by changing the value of the Input. But i don't want an input value to be a formatted number, I want to keep it in the original format that is common and suitable for API calls.

Examples:

Display value in English (Europe): 1,000,000.25
Display value in Serbia: 1.000.000,25
Value: 1000000.25

Display value should be only shown on blur event. The original value should be restored when typing but still supporting a decimal separator per locale.

### Proposed Solution

Introduce a display value that can be used for such behavior, that does not have to be equal to the input value.

### Priority

Medium

### Privacy Policy

- [X] I’m not disclosing any internal or sensitive information.

**Comment by unazko** [comment]
Hello @vasicvuk,

Perhaps you want an numeric input field similar to those in the example:
https://ui5.sap.com/#/entity/sap.m.StepInput/sample/sap.m.sample.StepInput

![NumericInput](https://github.com/SAP/ui5-webcomponents/assets/15860295/c1463766-66a4-477f-9c95-e32c7a94b8ce)

In this case that will be a `ui5-step-input` component improvement request.
Could you please test at the provided page if the component there behaves as expected and the I can forward the incident to the corresponding team.

Best regards,
Boyan

**Comment by vasicvuk** [comment]
Hi @unazko I guess the thousand separator can be implemented on both components. I don't need a step option for my case

**Comment by unazko** [comment]
Hi @vasicvuk,

So just to be sure. Your request is for a thousand separator in the `ui5-input` component right?

Best regards,
Boyan

**Comment by unazko** [comment]
Hello @SAP/ui5-webcomponents-topic-rl,

Please have a look at this feature request.

Best regards,
Boyan

**Comment by vasicvuk** [comment]
@unazko Right, but this option can be useful for step-input as you mentioned

**Comment by unazko** [comment]
Hello @SAP/ui5-webcomponents-topic-rl,

Please take a look at this feature request.

Best regards,
Boyan
